File: sdks/ts/src/sdk/models/shared/retentionpolicyassignmentbase.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum RetentionPolicyAssignmentBaseTypeEnum {
    RetentionPolicyAssignment = "retention_policy_assignment"
}


// RetentionPolicyAssignmentBase
/** 
 * A base representation of a retention policy assignment.
**/
export class RetentionPolicyAssignmentBase extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: RetentionPolicyAssignmentBaseTypeEnum;
}

File: python-sdk/src/boxsdk/models/shared/retentionpolicyassignment_base.py
import dataclasses
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class RetentionPolicyAssignmentBaseTypeEnum(str, Enum):
    RETENTION_POLICY_ASSIGNMENT = "retention_policy_assignment"


@dataclass_json
@dataclasses.dataclass
class RetentionPolicyAssignmentBase:
    r"""RetentionPolicyAssignmentBase
    A base representation of a retention policy assignment.
    """
    
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    type: RetentionPolicyAssignmentBaseTypeEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
